### PR TITLE
[push-voting-power] Ethereum Push Notification Service ($PUSH) Governance: Custom strategy for voting

### PR DESCRIPTION
**Pull request by mujtaba1747**
Changes proposed in this pull request:
- Adding a new strategy for PUSH governance

    This strategy weighs the vote using:
    - Amount of `$PUSH` delegated to user
    - Amount of `$PUSH` staked in Push staking pool
    - Amount of `$PUSH-LP` staked in LP staking pool
    
    The voting power is calculated in terms of `$PUSH`, so `$PUSH-LP` tokens are converted to `$PUSH` using on chain data from Uniswap-V2 Router, WETH and USDT token contracts.

### CODE REVIEW

File: src/strategies/push-voting-power/index.ts [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        internalType: "uint256[]",
+        name: "amounts",
+        type: "uint256[]"
+      }
+    ],
+    stateMutability: "view",
+    type: "function"
+  },
+
+];
```
Comment (ChaituVR) [inline src/strategies/push-voting-power/index.ts:65]:
```suggestion
const sharedABI = [
  'function balanceOf(address user, address token) view returns (uint256)',
  'function getAmountsOut(uint256 amountIn, address[] path) view returns (uint256[] amounts)'
];
```

File: src/strategies/push-voting-power/index.ts [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+      {
+        name: "",
+        type: "uint256"
+      }
+    ],
+    payable: false,
+    stateMutability: "view",
+    type: "function"
+  },
+];
```
Comment (ChaituVR) [inline src/strategies/push-voting-power/index.ts:87]:
```suggestion
const wethABI = [ 'function balanceOf(address) view returns (uint256)' ];
```

File: src/strategies/push-voting-power/index.ts [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+      {
+        internalType: "uint256",
+        name: "",
+        type: "uint256"
+      }
+    ],
+    stateMutability: "view",
+    type: "function"
+  },
+];
```
Comment (ChaituVR) [inline src/strategies/push-voting-power/index.ts:130]:
```suggestion
const epnsTokenABI = [
  'function getCurrentVotes(address account) view returns (uint96)',
  'function balanceOf(address account) view returns (uint256)'
];
```

File: src/strategies/push-voting-power/index.ts [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        internalType: "uint256",
+        name: "",
+        type: "uint256"
+      }
+    ],
+    payable: false,
+    stateMutability: "view",
+    type: "function"
+  }
+];
```
Comment (ChaituVR) [inline src/strategies/push-voting-power/index.ts:148]:
```suggestion
const epnsLpABI = [ 'function totalSupply() view returns (uint256)' ];
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Comment by mujtaba1747** [comment]
@ChaituVR, thanks for recommending the improvement!
I've made the changes, we can proceed further

**Review by mujtaba1747 — COMMENTED** [review]
Integrated the changes

**Comment by mujtaba1747** [comment]
@ChaituVR Any more changes needed from our side.
I made the changes, but I'm not sure why it isn't merging automatically

**Review by bonustrack — APPROVED** [review]

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**